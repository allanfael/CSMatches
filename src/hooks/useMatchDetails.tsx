import { DetailsRouteProps } from '@navigator/ParamsRoute'
import { useRoute } from '@react-navigation/native'
import { MatchDetail } from 'interfaces/match-detail'

export type UseMatchDetailsProps = {
  match: MatchDetail
}

export const useMatchDetails = (): UseMatchDetailsProps => {
  const { params } = useRoute<DetailsRouteProps>()

  return {
    match: params.match,
  }
}
