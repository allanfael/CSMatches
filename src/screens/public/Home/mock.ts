import { Match } from 'interfaces/matches'

export const matchesMock: Match[] = [
  {
    id: '1',
    time: '2024-01-09T14:00:00Z',
    live: false,
    teamOne: {
      name: 'teamOne',
      image: '',
    },
    teamTwo: {
      name: 'teamTwo',
      image: '',
    },
    league: {
      name: 'leagueTest',
      image: '',
    },
  },
]
