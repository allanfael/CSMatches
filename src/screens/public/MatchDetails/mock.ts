import { MatchDetail } from 'interfaces/match-detail'

export const matchDetailsMock: MatchDetail = {
  teamOne: {
    name: 'Nixuh',
    image:
      'https://cdn.pandascore.co/images/team/image/131795/nixuhlogo_square.png',
  },
  teamTwo: {
    name: 'Goliath',
    image:
      'https://cdn.pandascore.co/images/team/image/126559/goliath_gaming.png',
  },
  time: '2024-01-09T14:00:00Z',
  league: 'PGL Major',
  teamOnePlayers: [
    {
      name: 'Alexey',
      image:
        'https://cdn.pandascore.co/images/player/image/22702/900px_nickel_back_at_dh_dehli_2019.png',
      nickname: 'NickelBack',
    },
    {
      name: 'Vladislav',
      image:
        'https://cdn.pandascore.co/images/player/image/23232/600px_krad_at_champions_cup_finals.png',
      nickname: 'Krad',
    },
  ],
  teamTwoPlayers: [
    {
      name: 'Aleksandr',
      image: 'https://cdn.pandascore.co/images/player/image/24697/p24809.png',
      nickname: 'TRAVIS',
    },
    {
      name: 'Igor',
      image:
        'https://cdn.pandascore.co/images/player/image/30826/600px_forester_at_champions_cup_finals.png',
      nickname: 'Forester',
    },
  ],
}
