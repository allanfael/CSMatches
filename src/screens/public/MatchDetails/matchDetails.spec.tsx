import { useMatchDetails, UseMatchDetailsProps } from '@hooks/useMatchDetails'
import { render, screen } from '@testing-library/react-native'

import { matchDetailsMock } from './mock'
import { MatchDetails } from '.'

jest.mock('../../../hooks/useMatchDetails', () => ({
  useMatchDetails: jest.fn(),
}))

describe('<MatchDetails />', () => {
  let sut: JSX.Element

  beforeAll(() => {
    sut = <MatchDetails />
  })

  it('should render screen correctly', async () => {
    ;(
      useMatchDetails as jest.MockedFunction<typeof useMatchDetails>
    ).mockReturnValue({
      match: matchDetailsMock,
    } as UseMatchDetailsProps)

    render(sut)

    expect(screen.toJSON()).toMatchSnapshot()
  })

  it('should render screen with values correctly', async () => {
    ;(
      useMatchDetails as jest.MockedFunction<typeof useMatchDetails>
    ).mockReturnValue({
      match: matchDetailsMock,
    } as UseMatchDetailsProps)

    const { getByText } = render(sut)

    const teamOneName = getByText('Nixuh')
    const teamTwoName = getByText('Goliath')

    expect(teamOneName).toBeTruthy()
    expect(teamTwoName).toBeTruthy()
  })
})
