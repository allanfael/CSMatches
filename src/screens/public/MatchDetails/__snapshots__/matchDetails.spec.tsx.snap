// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchDetails /> should render screen correctly 1`] = `
<View
  style={
    {
      "marginTop": 0,
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View>
        <View>
          <View>
            <View>
              <View>
                <ViewManagerAdapter_ExpoImage
                  accessibilityLabel="image"
                  backgroundColor={4291085508}
                  contentFit="contain"
                  contentPosition={
                    {
                      "left": "50%",
                      "top": "50%",
                    }
                  }
                  onError={[Function]}
                  onLoad={[Function]}
                  onLoadStart={[Function]}
                  onProgress={[Function]}
                  placeholder={[]}
                  source={
                    [
                      {
                        "uri": "https://cdn.pandascore.co/images/team/image/131795/nixuhlogo_square.png",
                      },
                    ]
                  }
                  style={
                    {
                      "backgroundColor": "#C4C4C4",
                    }
                  }
                  transition={null}
                />
              </View>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Roboto_400Regular",
                    "fontSize": 10,
                    "lineHeight": 16,
                  }
                }
              >
                Nixuh
              </Text>
            </View>
            <Text
              style={
                {
                  "color": "#C4C4C4",
                  "fontFamily": "Roboto_400Regular",
                  "fontSize": 12,
                  "lineHeight": 18,
                }
              }
            >
              vs
            </Text>
            <View>
              <View>
                <ViewManagerAdapter_ExpoImage
                  accessibilityLabel="image"
                  backgroundColor={4291085508}
                  contentFit="contain"
                  contentPosition={
                    {
                      "left": "50%",
                      "top": "50%",
                    }
                  }
                  onError={[Function]}
                  onLoad={[Function]}
                  onLoadStart={[Function]}
                  onProgress={[Function]}
                  placeholder={[]}
                  source={
                    [
                      {
                        "uri": "https://cdn.pandascore.co/images/team/image/126559/goliath_gaming.png",
                      },
                    ]
                  }
                  style={
                    {
                      "backgroundColor": "#C4C4C4",
                    }
                  }
                  transition={null}
                />
              </View>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Roboto_400Regular",
                    "fontSize": 10,
                    "lineHeight": 16,
                  }
                }
              >
                Goliath
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 12,
              "lineHeight": 18,
            }
          }
        >
          Today at 11:00 AM
        </Text>
      </View>
      <View>
        <View>
          <View>
            <View>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Roboto_500Medium",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                NickelBack
              </Text>
              <Text
                style={
                  {
                    "color": "#C4C4C4",
                    "fontFamily": "Roboto_400Regular",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                Alexey
              </Text>
            </View>
            <View>
              <ViewManagerAdapter_ExpoImage
                accessibilityLabel="image"
                backgroundColor={4291085508}
                contentFit="cover"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                source={
                  [
                    {
                      "uri": "https://cdn.pandascore.co/images/player/image/22702/900px_nickel_back_at_dh_dehli_2019.png",
                    },
                  ]
                }
                style={
                  {
                    "backgroundColor": "#C4C4C4",
                  }
                }
                transition={null}
              />
            </View>
          </View>
          <View>
            <View>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Roboto_500Medium",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                Krad
              </Text>
              <Text
                style={
                  {
                    "color": "#C4C4C4",
                    "fontFamily": "Roboto_400Regular",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                Vladislav
              </Text>
            </View>
            <View>
              <ViewManagerAdapter_ExpoImage
                accessibilityLabel="image"
                backgroundColor={4291085508}
                contentFit="cover"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                source={
                  [
                    {
                      "uri": "https://cdn.pandascore.co/images/player/image/22702/900px_nickel_back_at_dh_dehli_2019.png",
                    },
                  ]
                }
                style={
                  {
                    "backgroundColor": "#C4C4C4",
                  }
                }
                transition={null}
              />
            </View>
          </View>
        </View>
        <View>
          <View>
            <View>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Roboto_500Medium",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                TRAVIS
              </Text>
              <Text
                style={
                  {
                    "color": "#C4C4C4",
                    "fontFamily": "Roboto_400Regular",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                Aleksandr
              </Text>
            </View>
            <View>
              <ViewManagerAdapter_ExpoImage
                accessibilityLabel="image"
                backgroundColor={4291085508}
                contentFit="cover"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                source={
                  [
                    {
                      "uri": "https://cdn.pandascore.co/images/player/image/22702/900px_nickel_back_at_dh_dehli_2019.png",
                    },
                  ]
                }
                style={
                  {
                    "backgroundColor": "#C4C4C4",
                  }
                }
                transition={null}
              />
            </View>
          </View>
          <View>
            <View>
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "Roboto_500Medium",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                Forester
              </Text>
              <Text
                style={
                  {
                    "color": "#C4C4C4",
                    "fontFamily": "Roboto_400Regular",
                    "fontSize": 14,
                    "lineHeight": 18,
                  }
                }
              >
                Igor
              </Text>
            </View>
            <View>
              <ViewManagerAdapter_ExpoImage
                accessibilityLabel="image"
                backgroundColor={4291085508}
                contentFit="cover"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                source={
                  [
                    {
                      "uri": "https://cdn.pandascore.co/images/player/image/22702/900px_nickel_back_at_dh_dehli_2019.png",
                    },
                  ]
                }
                style={
                  {
                    "backgroundColor": "#C4C4C4",
                  }
                }
                transition={null}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
